octobot/updater/python_updater.py

```diff
@@ -0,0 +1,123 @@
+#  Drakkar-Software OctoBot
+#  Copyright (c) Drakkar-Software, All rights reserved.
+#
+#  This library is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU Lesser General Public
+#  License as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+import json
+import os
+import subprocess
+from shlex import quote as shlex_quote
+
+import aiohttp
+import sys
+from packaging.version import parse
+
+import octobot.constants as constants
+import octobot.updater.updater as updater_class
+
+
+class PythonUpdater(updater_class.Updater):
+    def __init__(self):
+        super().__init__()
+        self.use_git = os.path.isdir(".git")
+
+    async def get_latest_version(self):
+        if self.use_git:
+            try:
+                self._run_git_fetch()
```

Comment:
should we quickly check if the user has git and if not log an error message saying git is required with a link to download it ?

Reply:
You mean the user has a ".git" folder without having git installed on his computer ?

Reply:
Right, I guess we can try without check